compiler/rustc_middle/src/mir/mono.rs

```diff
@@ -526,44 +525,53 @@ impl<'tcx> CodegenUnit<'tcx> {
         tcx: TyCtxt<'tcx>,
     ) -> Vec<(MonoItem<'tcx>, MonoItemData)> {
         // The codegen tests rely on items being process in the same order as
-        // they appear in the file, so for local items, we sort by node_id first
+        // they appear in the file, so for local items, we sort by span and
+        // def_path first
         #[derive(PartialEq, Eq, PartialOrd, Ord)]
-        struct ItemSortKey<'tcx>(Option<usize>, SymbolName<'tcx>);
-
+        struct ItemSortKey<'tcx>(Option<Span>, Option<String>, SymbolName<'tcx>);
+
+        // We only want to take HirIds of user-defines instances into account.
+        // The others don't matter for the codegen tests and can even make item
+        // order unstable.
+        fn local_item_query<'tcx, T>(
+            item: MonoItem<'tcx>,
+            op: impl FnOnce(DefId) -> T,
+        ) -> Option<T> {
```

Comment:
Can this function just return `Option<DefId>`, and let the caller call `.map`?

Reply:
As long as you don't use the DefId as sorting key, having it as a local variable is fine.